client/state/themes/actions/activate-style-variation.ts

```diff
@@ -0,0 +1,50 @@
+import { isDefaultGlobalStylesVariationSlug } from '@automattic/design-picker';
+import {
+	getGlobalStylesId,
+	getGlobalStylesVariations,
+	updateGlobalStyles,
+} from 'calypso/state/global-styles/actions';
+import { getCanonicalTheme } from 'calypso/state/themes/selectors';
+import type { GlobalStyles } from '@automattic/data-stores';
+import type { CalypsoDispatch } from 'calypso/state/types';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Triggers a network request to activate a specific style variation on a given site.
+ * @param {string}  themeId            Theme ID
+ * @param {number}  siteId             Site ID
+ * @param {string}  styleVariationSlug The slug of the style variation
+ * @returns {Function}                 Action thunk
+ */
+export function activateStyleVariation(
+	themeId: string,
+	siteId: number,
+	styleVariationSlug: string
+) {
+	return async ( dispatch: CalypsoDispatch, getState: AppState ) => {
+		const state = getState();
+		const theme = getCanonicalTheme( state, siteId, themeId );
+		const themeStylesheet = theme?.stylesheet || themeId;
+
+		if ( styleVariationSlug ) {
+			let currentVariation;
+
+			// Clear the global styles if the default style variation is selected.
+			if ( isDefaultGlobalStylesVariationSlug( styleVariationSlug ) ) {
+				currentVariation = {} as GlobalStyles;
+			} else {
+				const variations = await dispatch( getGlobalStylesVariations( siteId, themeStylesheet ) );
```

Comment:
This will add yet another endpoint call overhead. The current live preview already takes a long time... so I'd not recommend this 😢 we already have the style variation object so why not just use that.

Reply:
The style variations from themes on the Theme Showcase don't have `styles` (it's `{}`)... so we have to fetch a single theme or style variations again 🥲

Reply:
Ah.... I see...

In that case, maybe can fetch the variations just in the theme showcase (and not when it's called in theme details). Similar reason here: https://github.com/Automattic/wp-calypso/pull/85945#discussion_r1446097268

Reply:
> In that case, maybe can fetch the variations just in the theme showcase

A possible way is to return the complete variations from the `/themes` endpoint but it seems too heavy. Or we can fetch variations when people open the three-dot menu.